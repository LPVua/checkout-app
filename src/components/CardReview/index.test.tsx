import * as React from 'react'
import {render, cleanup} from '@testing-library/react';
import { CardReview } from '.';

describe('CardReview', () => {
  afterEach(cleanup);

  it('should render proper markup', () => {
    const { asFragment } = render(<CardReview rating={2} description="some description" title="some title" subtitle="some subtitle" />)
    
    expect(asFragment()).toMatchSnapshot();
  })

  it('should render proper markup with extra class', () => {
    const { asFragment } = render(<CardReview rating={2} description="some description" title="some title" subtitle="some subtitle" className="some-class"/>)
    
    expect(asFragment()).toMatchSnapshot();
  })
});
