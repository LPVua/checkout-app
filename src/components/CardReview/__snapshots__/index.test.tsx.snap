// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardReview should render proper markup 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.4rem;
  padding-left: .6rem;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  width: 100%;
}

.emotion-2 {
  font-weight: bold;
  margin-bottom: .3rem;
}

.emotion-3 {
  font-size: 1.4rem;
  color: var(--grey-2);
}

.emotion-6 {
  margin-top: .6rem;
}

.emotion-1 {
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.8rem;
  padding: 0 0.6rem;
  border-radius: .3rem;
  background: #1abc9c;
  color: white;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: var(--red-1);
  margin-right: .6rem;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-1"
      >
        2 / 
        <span
          class="emotion-0"
        >
          5
        </span>
      </div>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-2"
        >
          some title
        </div>
        <div
          class="emotion-3"
        >
          some subtitle
        </div>
      </div>
    </div>
    <div
      class="emotion-6"
    >
      some description
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardReview should render proper markup with extra class 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.4rem;
  padding-left: .6rem;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  width: 100%;
}

.emotion-2 {
  font-weight: bold;
  margin-bottom: .3rem;
}

.emotion-3 {
  font-size: 1.4rem;
  color: var(--grey-2);
}

.emotion-6 {
  margin-top: .6rem;
}

.emotion-1 {
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.8rem;
  padding: 0 0.6rem;
  border-radius: .3rem;
  background: #1abc9c;
  color: white;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: var(--red-1);
  margin-right: .6rem;
}

<div
    class="emotion-7 some-class"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-1"
      >
        2 / 
        <span
          class="emotion-0"
        >
          5
        </span>
      </div>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-2"
        >
          some title
        </div>
        <div
          class="emotion-3"
        >
          some subtitle
        </div>
      </div>
    </div>
    <div
      class="emotion-6"
    >
      some description
    </div>
  </div>
</DocumentFragment>
`;
